http/interceptor.go

```diff
@@ -67,9 +57,7 @@ type ResponseWriter interface {
 // the http interfaces implemented by the original response writer to avoid
 // the observer effect.
 // Heavily inspired in https://github.com/openzipkin/zipkin-go/blob/master/middleware/http/server.go#L218
-func wrap(w http.ResponseWriter, r *http.Request, tx types.Transaction) http.ResponseWriter { // nolint:gocyclo
-	i := &rwInterceptor{w: w, tx: tx, proto: r.Proto}
-
+func (i *rwInterceptor) wrap(w http.ResponseWriter, r *http.Request, tx types.Transaction) http.ResponseWriter { // nolint:gocyclo
```

Comment:
Curious about this change. I think given the code in https://github.com/corazawaf/coraza/pull/460/files#diff-8c63f2ea8fc5dc64c72d3df8ea920ef1e263bf6d98154934846f545030b31b23R43 I would keep this as a constructor function instead of middleware dealing with internals. For example, in https://github.com/corazawaf/coraza/pull/460/files#diff-8c63f2ea8fc5dc64c72d3df8ea920ef1e263bf6d98154934846f545030b31b23R43 you pass the proto but here you just passed the request. Also if Proto is passed on construction no need to pass request in here.

Reply:
Shall we remove the `request` from this signature?

Reply:
You are right! Done